test/unit/rules/require-input-label-test.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+const generateRuleTests = require('../../helpers/rule-test-harness');
+const ERROR_MESSAGE = require('../../../lib/rules/require-input-label').ERROR_MESSAGE;
+
+generateRuleTests({
+  name: 'require-input-label',
+
+  config: true,
+
+  good: [
+    '<label>LabelText<input /></label>',
+    '<label><input />LabelText</label>',
+    '<label>LabelText<Input /></label>',
+    '<label><Input />LabelText</label>',
+    '<label><div><input /></div></label>', // technically okay, hopefully no one does this though
+    '<input id="probablyHasLabel" />', // it's likely to have an associated label if it has an id attribute
+    '<input aria-label={{labelText}} />',
+    '<input aria-labelledby="someIdValue" />',
+    '<div></div>',
+    '<input ...attributes/>', // we are unable to correctly determine if this has a label or not, so we have to allow it
+  ],
+
+  bad: [
```

Comment:
Please add additional "bad" examples for each of the following:

A label wrapping an input, but without label text:

```hbs
<label><input></label>
```

Usage of `{{input}}`:

```hbs
{{input}}
```

```hbs
<div>{{input}}</div>
```

Usage of `<Input/>`:

```hbs
<Input />
```